### PR TITLE
Add parsing support for `ANY`/`SOME`/`ALL` comparison modifiers.

### PR DESCRIPTION
**Pull request by arthurschreiber**
## Description

This PR introduces support for the `ANY` and `ALL` operators in subqueries within the Vitess SQL parser.
For the time being, it's only supported when the query can be merged into a single query that is passed down to MySQL.

### Example Usage

```sql
SELECT *
FROM employees
WHERE salary > ANY (SELECT salary FROM interns);

SELECT *
FROM employees
WHERE salary > ALL (SELECT salary FROM interns);
```

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by arthurschreiber** [comment]
Note: This currently allows using the `ANY`/`SOME`/`ALL` modifiers in combination with the null safe comparison operator (`<=>`) which is not actually allowed by MySQL.

This should probably be fixed so our parser matches MySQL's parser more closely.

File: go/vt/vtgate/planbuilder/testdata/unsupported_cases.json [hunk 1]
```diff
@@ -358,5 +358,10 @@
     "comment": "WITH ROLLUP not supported on sharded queries",
     "query": "select a, b, c, sum(d) from user group by a, b, c with rollup",
     "plan": "VT12001: unsupported: GROUP BY WITH ROLLUP not supported for sharded queries"
+  },
+  {
+    "comment": "WITH ROLLUP not supported on sharded queries",
```
Comment (dbussink) [inline go/vt/vtgate/planbuilder/testdata/unsupported_cases.json:363]: This comment seems off?

Comment (systay) [reply go/vt/vtgate/planbuilder/testdata/unsupported_cases.json:363]: image

**Comment by GuptaManan100** [comment]
I have pushed a fix now for the parser conflicts. I've also made the parser fail for queries that have null safe equality with a column modifier! 💕

**Review by systay — APPROVED** [review]

**Review by GuptaManan100 — APPROVED** [review]
LGTM

**Review by frouioui — APPROVED** [review]

**Pull request merged**